### PR TITLE
Translation Feature For The Module

### PR DESCRIPTION
**Pull request by progammer-rkt**
- Translations are kept under `src/reactapp/public/i18n` directory. There is an `en_US.json` file for reference
- Implemented a universal method `__()` to do the translation magic
- Depending upon the store and locale, we will be populating the data attribute `data-language` of the root div element. From this, react app will use the right locale json file from the `i18n` directory
- On build, we will copy all json files into the directory `src/view/frontend/web/js/i18n` so that we can serve those files statically.

### CODE REVIEW

File: src/reactapp/build/i18n/de_DE.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    "Postal Code": "Postleitzahl",
+    "Price": "Preis",
+    "Price: {}": "Preis: {}",
+    "Product Details": "Produktdetails",
+    "Qty": "Menge",
+    "Required": "Erforderlich",
+    "Shipping Information": "Versandinformationen",
+    "Shipping Methods": "Versandmethoden",
+    "Shipping address update failed. Please try again": "Aktualisierung der Versandadresse fehlgeschlagen. Bitte erneut versuchen",
+    "Shipping address updated successfully": "Versandanzeige Kleid erfolgreich aktualisiert",
```
Comment (schmengler) [inline src/reactapp/build/i18n/de_DE.json:46]: correct translation: "Versandadresse erfolgreich aktualisiert" (as below)

Comment (schmengler) [reply src/reactapp/build/i18n/de_DE.json:46]: (once with `.` and once without)

**Review by schmengler — APPROVED** [review]

File: src/reactapp/build/i18n/de_DE.json (hunk 1 above)
Comment (progammer-rkt) [reply src/reactapp/build/i18n/de_DE.json]: This is corrected 👍

**Review by jesse-deboer — CHANGES_REQUESTED** [review]
Looks really good, mostly small things added.

For the all-caps buttons like `UPDATE` I would recommend to just set it to `Update`. This is better for consistency in the translations. If you do want to use all caps for a button it's better to set the `uppercase` class instead.

Also, the data-language is fixed on default to `en_US`. Is this being updated by JS or is it maybe better to use a variable here?

**Comment by progammer-rkt** [comment]
> Also, the data language is fixed by default to `en_US`. Is this being updated by JS or is it maybe better to use a variable here?

I cannot see any added advantage of making this a variable. The file `src/reactapp/public/index.html` is the entry point of the react app. This file is in action only when you started the app as standalone, detaching from any Magento site. So with an `en_US.json` file that intends to hold all the translations and for the reference for all other language json files, the hard-coded `en_US` is just fine. 

If in case, someone wants a different locale in the app mode, all they need to do is, add the translation json file inside `public/i18n` directory and change the language reference in the DOM.

File: src/reactapp/src/components/address/components/ORBox.jsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import React from 'react';
+import { __ } from '../../../i18n';
 
 function ORBox() {
   return (
     <div className="flex items-center justify-center my-2 italic font-semibold">
-      OR
+      {__('OR')}
```
[the remark this answers is not in the document]
Comment (progammer-rkt) [reply src/reactapp/src/components/address/components/ORBox.jsx:7]: Let's keep this 'OR`, which is what visually appealing

**Pull request merged**